### DISCUSSION TITLE
🎉  CNCF is accepted to Google Summer of Code 2024

### DISCUSSION THREAD

**Discussion by aliok** [cncf/mentoring#1186] [Announcements]
Google Summer of Code has selected the CNCF as one of the mentoring organizations!

## Mentors

It is possible to add ideas until the 1 week before (March 11) the contributor application period start date (March 18).
We encourage communities to add ideas earlier rather than later to give potential contributors more time.

You can add your ideas to this document by sending a PR: https://github.com/cncf/mentoring/blob/main/programs/summerofcode/2024.md

For more information, please see the [Mentor Guide](https://google.github.io/gsocguides/mentor/).

## Contributor (mentee) candidates

We encourage contributors to explore project ideas and engage with mentors and communities.
Between now and March 18, Potential GSoC contributors can discuss application ideas with the projects.
After March 18, you need to send an application with your proposal on GSoC platform.

You can find this list of ideas here: https://github.com/cncf/mentoring/blob/main/programs/summerofcode/2024.md.

You can reach out on the proposal's upstream issue and project's communication channels to discuss the issue.

For more information, please see the [GSoC Contributor guide](https://google.github.io/gsocguides/student/).

## Important Upcoming Dates

For transparency here are all the dates that are coming up, for mentors, prospective contributors, and the CNCF org administrators.

For the full timeline, please see <https://developers.google.com/open-source/gsoc/timeline>,

**Now - March 18**: Mentors, proactive GSoC contributors will reach out to you to ask questions about [our proposals list](https://github.com/cncf/mentoring/blob/main/programs/summerofcode/2024.md) and receive feedback from you so they can start crafting their project proposals.

**March 18 - April 2 18:00 UTC**: GSoC contributors will submit their proposals through the program website for mentors to give solid feedback on.

**April 2 - 23**: Mentorship WG team to review all submitted GSoC Contributor proposals and consider how many to want to select and how many we can handle (how many we have committed mentors for). Decide on the ranking of the GSoC Contributor slots. At least 1 mentor must be assigned to each project before it can be selected.

**April 24 18:00 UTC**: Deadline to submit ranked slot requests (CNCF org Admins enter these requests)

**April 24 - May 1**: Google Program Admins review and assign org slots

**May 3**: Organizations receive notification of their accepted GSoC 2023 Contributors

**May 4**: Accepted GSoC 2023 GSoC Contributor projects are announced

**May 1 - 26**: Community Bonding Period

**May 27**: Coding begins

**Comment by alabulei1** [comment]
Great news! Check out WasmEdge's project ideas [here](https://github.com/cncf/mentoring/blob/main/programs/summerofcode/2024.md#wasmedge).

**Comment by abhayguptas** [comment]
Excited! 🙂

**Comment by aliok** [comment]
FYI, GSoC contributor (mentee) applications are now open: https://github.com/cncf/mentoring/discussions/1200

**Comment by AkihiroSuda** [comment]
> April 2 - 23: Mentorship WG team to review all submitted GSoC Contributor proposals and consider how many to want to select and how many we can handle (how many we have committed mentors for). Decide on the ranking of the GSoC Contributor slots. At least 1 mentor must be assigned to each project before it can be selected.


When and how will this happen this year? (Asking as a mentor for https://lima-vm.io)

**Reply by nate-double-u** [reply]
Google Summer of Code 2025's timeline puts the Proposal Deadline on April 8, and the Proposal Ranking Deadline on April 29th.
https://summerofcode.withgoogle.com/organizations/cncf/programs/2025/timeline

Once the proposals are in from the candidates, the mentors of each project proposal will be able to mark them as 'interested in,' then CNCF staff will make the final ranking from the proposals that the mentors are interested in working on. I'll be sending out a more detailed timeline in the next couple weeks as we will need to have some time to make sure we have time to let the mentors review as well as for the CNCF staff to have time to do the ranking.
